observ/store.py

```diff
@@ -0,0 +1,135 @@
+import copy
+from functools import partial, wraps
+from inspect import signature
+from typing import Callable, TypeVar
+
+from .dep import Dep
+from .observables import (
+    reactive,
+    readonly,
+    shallow_reactive,
+    to_raw,
+)
+from .watcher import Watcher
+
+
+T = TypeVar("T", bound=Callable)
+
+
+registry = {}
+
+
+def mutation(fn: T) -> T:
+    registry[fn] = "mutation"
+
+    @wraps(fn)
+    def inner(self, *args, **kwargs):
+        return fn(self, *args, **kwargs)
```

Comment:
If you're not altering the behavior in any way, you can skip all this and just `return fn` as is!

Reply:
I should have reviewed the code a bit better before pushing I guess :)